lib/net.c

```diff
@@ -720,10 +720,62 @@ get_if_back(struct net_config *net_conf)
 	ETH_RSS_NONFRAG_IPV6_OTHER | \
 	ETH_RSS_IPV6_EX)
 
+static int
+check_port_rss_key_update(struct gatekeeper_if *iface, uint16_t port_id,
+	bool *rss_key_updatable, struct rte_eth_rss_conf *rss_conf)
+{
+	struct rte_eth_rss_conf cur_rss_conf;
+	int ret = rte_eth_dev_rss_hash_update(port_id, rss_conf);
+	if (ret == 0)
+		return ret;
+
+	ret = rte_eth_dev_rss_hash_conf_get(port_id, &cur_rss_conf);
+	if (ret < 0) {
+		G_LOG(WARNING,
+			"net: the %s interface does not support a critical hardware capability: update and get the RSS hash configuration\n",
+			iface->name);
+		return ret;
+	}
+
+	/*
+	 * At least two ports with different RSS hash keys
+	 * cannot be updated.
+	 */
+	if (!*rss_key_updatable &&
+			(cur_rss_conf.rss_key == NULL ||
```

Comment:
We fail whenever a single NIC cannot update its RSS key, so variable `rss_key_updatable` is not doing anything here. So we can drop it altogether.